src/Microsoft.VisualStudio.Threading/AsyncLazy`1.cs

```diff
@@ -178,7 +178,6 @@ public Task<T> GetValueAsync(CancellationToken cancellationToken)
                         }
                         finally
                         {
-                            this.jobFactory = null;
```

Comment:
It might be a good idea to make `jobFactory` readonly to prevent this issue from cropping up again in the future.